cluster-autoscaler/cloudprovider/azure/azure_cache.go

```diff
@@ -353,6 +378,46 @@ func (m *azureCache) getAutoscalingOptions(ref azureRef) map[string]string {
 	return m.autoscalingOptions[ref]
 }
 
+// HasInstance returns if a given instance exists in the azure cache
+func (m *azureCache) HasInstance(providerID string) (bool, error) {
+	vmsPoolSet := m.getVMsPoolSet()
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+	resourceID, err := convertResourceGroupNameToLower(providerID)
+	if err != nil {
+		// Most likely an invalid resource id, we should return false
+		// most of these shouldn't make it here do to higher level
+		// validation in the HasInstance azure.cloudprovider function
+		return false, err
+	}
+
+	nodeGroupNameForNode := getVMSSNameFromProviderID(providerID)
+	inst := azureRef{Name: resourceID}
+	klog.V(4).Infof("HasInstance: Checking instance %s", inst.Name)
+	if m.unownedInstances[inst] || (m.vmType == vmTypeVMSS && len(vmsPoolSet) == 0 && !m.registeredAKSPoolNames.Has(nodeGroupNameForNode)) {
```

Comment:
Why we don't need this in `FindForInstance()`?

Reply:
We potentially could need this for FindForInstance. I am just limiting the blast radius of this change.